data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/stream/ShardConsumer.java

```diff
@@ -242,100 +241,88 @@ public void run() {
+                sequenceNumber = response.records().get(response.records().size() - 1).dynamodb().sequenceNumber();
+                Instant lastEventTime = response.records().get(response.records().size() - 1).dynamodb().approximateCreationDateTime();
 
-                GetRecordsResponse response = callGetRecords(shardIterator);
-                shardIterator = response.nextShardIterator();
-                if (!response.records().isEmpty()) {
-                    // Always use the last sequence number for checkpoint
-                    sequenceNumber = response.records().get(response.records().size() - 1).dynamodb().sequenceNumber();
-                    Instant lastEventTime = response.records().get(response.records().size() - 1).dynamodb().approximateCreationDateTime();
+                if (lastEventTime.isBefore(startTime)) {
+                    LOG.debug("Get {} events before start time, ignore...", response.records().size());
+                    continue;
+                }
+                if (waitForExport) {
+                    waitForExport();
+                    waitForExport = false;
+                }
 
-                    if (lastEventTime.isBefore(startTime)) {
-                        LOG.debug("Get {} events before start time, ignore...", response.records().size());
-                        continue;
-                    }
-                    if (waitForExport) {
-                        checkpointer.checkpoint(sequenceNumber);
-                        waitForExport();
-                        waitForExport = false;
-                    }
-                    records = response.records().stream()
-                            .filter(record -> record.dynamodb().approximateCreationDateTime().isAfter(startTime))
-                            .collect(Collectors.toList());
-                    recordConverter.writeToBuffer(acknowledgementSet, records);
-                    shardProgress.increment();
-                    recordsWrittenToBuffer += records.size();
-                    long delay = System.currentTimeMillis() - lastEventTime.toEpochMilli();
-                    interval = delay > GET_RECORD_DELAY_THRESHOLD_MILLS ? MINIMUM_GET_RECORD_INTERVAL_MILLS : GET_RECORD_INTERVAL_MILLS;
-
-                } else {
-                    interval = GET_RECORD_INTERVAL_MILLS;
-                    shardProgress.increment();
+                AcknowledgementSet acknowledgementSet = null;
+                if (shardAcknowledgementManager != null) {
+                    final StreamProgressState lastProgressState = streamPartition.getProgressState().orElseThrow();
+                    final String endingSequenceNumber = lastProgressState.getEndingSequenceNumber();
+                    acknowledgementSet = shardAcknowledgementManager.createAcknowledgmentSet(streamPartition, lastProgressState.getSequenceNumber(), endingSequenceNumber == null || endingSequenceNumber.isEmpty());
```

Comment:
Will the sequence number always be null for the first acknowledgment set we create since we haven't done any checkpointing there yet? Shouldn't the condition for the last partition be if shardIterator is null?